benchmark.py

```diff
@@ -40,16 +43,38 @@
 
 checkpoint = torch.load(args.checkpoint, map_location = 'cpu') if args.checkpoint else None
 if checkpoint:
-	args.model, args.lang, args.sample_rate, args.window_size, args.window_stride, args.window, args.num_input_features = map(checkpoint['args'].get, ['model', 'lang', 'sample_rate', 'window_size', 'window_stride', 'window', 'num_input_features'])
+	args.model, args.lang, args.sample_rate, args.window_size, args.window_stride, args.window, args.num_input_features = map(
+		checkpoint['args'].get,
+		['model', 'lang', 'sample_rate', 'window_size', 'window_stride', 'window', 'num_input_features'])
 
 use_cuda = 'cuda' in args.device
 
 labels = datasets.Labels(datasets.Language(args.lang))
 
+if args.profile_cuda:
+	os.environ["CUDNN_LOGINFO_DBG"] = '1'
+	os.environ["CUBLAS_LOGINFO_DBG"] = '1'
+
 if args.onnx:
 	onnxruntime_session = onnxruntime.InferenceSession(args.onnx)
-	model = lambda x: onnxruntime_session.run(None, dict(x = x))
-	load_batch = lambda x: x.numpy()
+
+	def load_batch_to_device(batch, device):
+		batch = batch.numpy()
+		io_binding = onnxruntime_session.io_binding()
+		batch_ortvalue = onnxruntime.OrtValue.ortvalue_from_numpy(batch, device, 0)
+		io_binding.bind_input(
+			name = 'x',
+			device_type = batch_ortvalue.device_name(),
+			device_id = 0,
+			element_type = batch.dtype,
+			shape = batch_ortvalue.shape(),
+			buffer_ptr = batch_ortvalue.data_ptr()
+		)
+		io_binding.bind_output(args.onnx_output_node_name, device)
+		return io_binding
+
+	load_batch = lambda batch: load_batch_to_device(batch, args.device)
```

Comment:
again, why do perform binding set up and CPU->GPU copy at every batch processing. let's do change benchmark.py to the same scheme we used in your other benchmark: move the batch to GPU before the loop and then just measure the GPU computation part.

Reply:
fixed